docs/snapshotters/erofs.md

```diff
@@ -0,0 +1,143 @@
+# EROFS Snapshotter
+
+The [EROFS](https://erofs.docs.kernel.org) snapshotter is an experimental
+feature, which is able to leverage EROFS-formatted blobs for each committed
+snapshot and prepares an EROFS + OverlayFS mount for each active snapshot.
+
+In order to leverage EROFS-formatted blobs, the EROFS differ is needed to be
+used together to apply image layers.  Otherwise, the EROFS snapshotter will
+just behave as the existing OverlayFS snapshotter: the default applier will
+unpack the image layer into the active EROFS snapshot, and commit it.
+
+Although it sounds somewhat similar to an enhanced OverlayFS snapshotter but
+I believe there are clear differences if looking into `s.mount()` and it highly
+tightens to the EROFS internals.  Currently, it's not quite clear to form an
+enhanced OverlayFS snapshotter directly, and (I think) it's not urgent since
+in the very beginning, it'd be better to be left as an independent snapshotter
+so that existing overlayfs users won't be impacted by the new behaviors and
+users could have a chance to try and develop the related ecosystems (such as
+ComposeFS, confidential containers, gVisor, Kata, gVisor, and more) together.
```

Comment:
I'm interested in the "confidential containers" case. How the guest would know the snapshotter prepared everything correctly (and the image integrity is preserved)?

Reply:
I think some changes should be made in the kata-shim, currently I don't have enough time to prepare a optimized workable implementation for this, but after this lands, I will try to sort out these.